src/main/java/org/aksw/simba/squirrel/analyzer/impl/GuessMimeType.java

```diff
@@ -0,0 +1,51 @@
+package org.aksw.simba.squirrel.analyzer.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import org.aksw.simba.squirrel.analyzer.Guess;
+import org.apache.tika.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GuessMimeType implements Guess {
+	private static final Logger LOGGER = LoggerFactory.getLogger(RDFAnalyzer.class);
+
+	public String guess(File data) {
+		String mimeType = null;
+		try {
+			FileInputStream inputStream = new FileInputStream(data.getAbsolutePath());
+			String fileContent = IOUtils.toString(inputStream);

```

Comment:
Loading the file completely into the memory is not a good idea. Based on this solution, you should implement an automaton that reads only the first bytes until it has decided which mime type it has seen.